python/mp/effects/effect.py

```diff
@@ -22,13 +22,28 @@ class Effect(abc.ABC):
     dm = DispatcherMapper()
 
     def __init__(self, *args, **kwargs):
+        # id will be assigned when the effect is attached to the mainloop
+        self.id = None
 
         # Introducing...the option to pass a rosary on init instead of
         # assigning it afterwards
         self.rosary = kwargs.get('rosary')
+
         # the name is used when the Effect is registered
         self.name = kwargs.get('name')
 
+        self.sort_registry = {
+            'cw': self.bead_set_sort_cw,
+            'ccw': self.bead_set_sort_ccw
+        }
+
+        # the sorting function for beads can be passed as an argument
+        self.bead_set_sort = kwargs.get('bead_set_sort', self.bead_set_sort_cw)
+        if (type(self.bead_set_sort) == str):
+            # default to cw
```

Comment:
This is so you can pass either a string (name) or a function as the argument to bead_set_sort